frame/alliance/src/lib.rs

```diff
@@ -0,0 +1,1089 @@
+//! - `veto` - Veto on a motion about `set_rule` and `elevate_ally`.
+//!
+//! #### For Super Users
+//! - `init_founders` - Initialize the founding members.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+mod types;
+pub mod weights;
+
+use sp_runtime::{
+	traits::{StaticLookup, Zero},
+	RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+use frame_support::{
+	codec::{Decode, Encode},
+	dispatch::{
+		DispatchError, DispatchResult, DispatchResultWithPostInfo, Dispatchable, GetDispatchInfo,
+		PostDispatchInfo,
+	},
+	ensure,
+	scale_info::TypeInfo,
+	traits::{
+		ChangeMembers, Currency, Get, InitializeMembers, IsSubType, OnUnbalanced,
+		ReservableCurrency,
+	},
+	weights::{Pays, Weight},
+};
+
+pub use pallet::*;
+pub use types::*;
+pub use weights::*;
+
+/// Simple index type for proposal counting.
+pub type ProposalIndex = u32;
+
+type Url = Vec<u8>;
+
+type BalanceOf<T, I = ()> =
+	<<T as Config<I>>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+type NegativeImbalanceOf<T, I = ()> = <<T as Config<I>>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+pub trait IdentityVerifier<AccountId> {
+	fn has_identity(who: &AccountId, fields: u64) -> bool;
+
+	fn has_good_judgement(who: &AccountId) -> bool;
+
+	fn super_account_id(who: &AccountId) -> Option<AccountId>;
+}
+
+impl<AccountId> IdentityVerifier<AccountId> for () {
+	fn has_identity(_who: &AccountId, _fields: u64) -> bool {
+		true
+	}
+
+	fn has_good_judgement(_who: &AccountId) -> bool {
+		true
+	}
+
+	fn super_account_id(_who: &AccountId) -> Option<AccountId> {
+		None
+	}
+}
+
+pub trait ProposalProvider<AccountId, Hash, Proposal> {
+	fn propose_proposal(
+		who: AccountId,
+		threshold: u32,
+		proposal: Box<Proposal>,
+		length_bound: u32,
+	) -> Result<(u32, u32), DispatchError>;
+
+	fn vote_proposal(
+		who: AccountId,
+		proposal: Hash,
+		index: ProposalIndex,
+		approve: bool,
+	) -> Result<bool, DispatchError>;
+
+	fn veto_proposal(proposal_hash: Hash) -> u32;
+
+	fn close_proposal(
+		proposal_hash: Hash,
+		index: ProposalIndex,
+		proposal_weight_bound: Weight,
+		length_bound: u32,
+	) -> DispatchResultWithPostInfo;
+
+	fn proposal_of(proposal_hash: Hash) -> Option<Proposal>;
+}
+
+/// The role of members.
+#[derive(Copy, Clone, PartialEq, Eq, RuntimeDebug, Encode, Decode, TypeInfo)]
+pub enum MemberRole {
+	Founder,
+	Fellow,
+	Ally,
+}
+
+/// The item type of blacklist.
+#[derive(Clone, PartialEq, Eq, RuntimeDebug, Encode, Decode, TypeInfo)]
+pub enum BlacklistItem<AccountId> {
+	AccountId(AccountId),
+	Website(Url),
+}
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	#[pallet::without_storage_info]
```

Comment:
Can the pallet be modified to make this storage bounded?

Reply:
updated the pallet. will update the runtime & mock later